### ISSUE TITLE
readgroup parsing with -R returns only a subset of reads

### ISSUE THREAD

**Issue by chrisamiller** [brentp/mosdepth#153]
The linked test cram below contains two readgroups, `2904129095` and `2904139476`, with two reads from each on chr1 and chr2.  

If I pull the latest quay.io release of mosdepth `quay.io/biocontainers/mosdepth:0.3.1--h01d7912_2` and then run mosdepth using per-readgroup options that should include all reads, I only get output from chromosome 1

```
$ mosdepth -R 2904129095,2904139476 -b 10000 -n test_rg -f all_sequences.fa test.cram

$ zgrep -v "0.00$" test_rg.regions.bed.gz
chr1	10000	20000	0.02

```

If I run it using no readgroups, I see the expected output:

```
mosdepth -b 10000 -n test_norg -f all_sequences.fa test.cram

$ zgrep -v "0.00$" test_norg.regions.bed.gz
chr1	10000	20000	0.02
chr2	10000	20000	0.01
```

The same problem persists when I give it a single readgroup, which is the expected usage of the feature (to subset a cram on the fly).

```
$ mosdepth -R 2904129095 -b 10000 -n test_rg_95 -f all_sequences.fa test.cram

$ zgrep -v "0.00$" test_rg_95.regions.bed.gz
chr1	10000	20000	0.02
```

Am I missing something obvious here or is this a bug?

Test data: [cram file](https://wustl.box.com/s/mor0fznu3ttgr5kkff6wqifyrrl1topo) and [index](https://wustl.box.com/s/ti0umqe31o9f7kwjby27cpjsntst9gtr)
Uses a GRCh38 fasta reference, but I can attach that as well if needed.

**Comment by brentp** [comment]
this is weird. i can recreate the bug, but it goes away when i build mosdepth in debug mode. i'll have a closer look this weekend.

**Comment by brentp** [comment]
This is fixed in v0.3.2: https://github.com/brentp/mosdepth/releases/tag/v0.3.2
thanks very much for the clear test-case. (I think not too many people are using read-groups)

**Comment by chrisamiller** [comment]
Thanks, Brent!  Had to make the test case anyway to convince myself that I wasn't crazy :)
